app/controllers/benefits/CompanyBenefitController.scala

```diff
@@ -73,11 +74,15 @@ trait CompanyBenefitController extends TaiBaseController
             } yield {
               employment match {
                 case Some(employment) =>
-                  val cache = Map(EndCompanyBenefit_EmploymentNameKey -> employment.name)
+
+                  val formattedBenefitName = (raw"(B|b)enefit(s)?".r replaceAllIn(
+                    Messages(s"tai.taxFreeAmount.table.taxComponent.${currentCache(EndCompanyBenefit_BenefitTypeKey)}"),"")).trim
```

Comment:
I think view logic should be in view, viewModels or viewHelpers depending on the situation, but not in controllers